src/Adapter/Cart/CommandHandler/UpdateCartAddressesHandler.php

```diff
@@ -75,14 +75,10 @@ public function handle(UpdateCartAddressesCommand $command)
      */
     private function fillCartWithCommandData(Cart $cart, UpdateCartAddressesCommand $command): void
     {
-        if ($command->getNewDeliveryAddressId()) {
```

Comment:
I don't understand why you remove the `if` and force the usecase. It changes the logic? 🤔

Reply:
As `getNewDeliveryAddressId` returns always an object, the `if` is unuseful.

Reply:
That looks like a mistake that you have found.
From the code we can see it's supposed to update either the delivery address or the invoice address.

ping @jolelievre wdyt?

Reply:
It's a bug. It should be fixed in an another PR. That's not the goal of this PR.

Reply:
@Progi1984 It depends. If there are some edge-cases where `$command->getNewDeliveryAddressId()` can be null, and the logic depends on it, then this PR will break this behavior.

Sometimes buggy code introduce a behavior which becomes used (the "it's not a feature it's a bug" joke 😄 ) I'm just trying to make sure we are not in this (sorry) usecase. Especially with Cart logic, it's important to be careful when dealing with this logic.

Reply:
Actually, this PR doesn't break anything.  `getNewDeliveryAddressId` can't be null. Check `UpdateCartAddressesCommand`.